sonic-net/sonic-buildimage
# [QoS] Add tunnel pipe mode support for IPIP Decap mode to use SAI_TUNNEL_ATTR_DECAP_QOS_DSCP_TO_TC_MAP

#### Why I did it
Support for pipe mode in IPIP packets when decapped - request from NetScan. Current support is for all platforms except Mellanox.

##### Work item tracking
- Microsoft ADO **(number only)**: 27565432

#### How I did it
Add key decap_dscp_to_tc_map to point to AZURE mapping in the DSCP_TO_TC_MAP table
#### How to verify it
Run `test_qos_dscp_to_tc_mapping.py` and see that it passes in pipe mode. Also check swss logs to see SAI_TUNNEL_ATTR_DECAP_QOS_DSCP_TO_TC_MAP attribute being used.



#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)
20231130



#### Description for the changelog

Change IPIP Decap Tunnel Mode to Pipe Mode from Uniform Mode


#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
